--- a/joern-cli/src/universal/joern-scan
+++ b/joern-cli/src/universal/joern-scan
@@ -15,6 +15,9 @@
 
 echo "Writing logs to: /tmp/joern-scan-log.txt"
 $SCRIPT -J-XX:+UseG1GC -J-XX:CompressedClassSpaceSize=128m -Dlog4j.configurationFile="$SCRIPT_ABS_DIR"/conf/log4j2.xml -J-XX:+UseStringDeduplication "$@" 2> /tmp/joern-scan-log.txt
-if [ $? -eq 2 ]; then
+status=$?
+if [ $status -eq 2 ]; then
   $SCRIPT -J-XX:+UseG1GC -J-XX:CompressedClassSpaceSize=128m -Dlog4j.configurationFile="$SCRIPT_ABS_DIR"/conf/log4j2.xml -J-XX:+UseStringDeduplication "$@" 2> /tmp/joern-scan-log.txt
+  status=$?
 fi
+exit $status
